paasta_tools/cli/cmds/wait_for_deployment.py

```diff
@@ -0,0 +1,323 @@
+    cluster_map = get_cluster_instance_map_for_service(soa_dir,
+                                                       service,
+                                                       deploy_group)
+    if not cluster_map:
+        _log(service=service,
+             component='deploy',
+             line=("Couldn't find any instances for service {0} in"
+                   " deploy group {1}").format(service, deploy_group),
+             level='event')
+        raise NoInstancesFound
+
+    print("Waiting for deployment of {0} for '{1}' complete..."
+          .format(git_sha, deploy_group))
+
+    for cluster in cluster_map.values():
+        cluster['deployed'] = 0
+    try:
+        with Timeout(seconds=timeout):
+            total_instances = sum([len(v["instances"])
+                                   for v in cluster_map.values()])
+            with progressbar.ProgressBar(maxval=total_instances) as bar:
+                while True:
+                    for cluster, instances in cluster_map.items():
+                        if cluster_map[cluster]['deployed'] != len(cluster_map[cluster]['instances']):
+                            cluster_map[cluster]['deployed'] = instances_deployed(
+                                cluster=cluster,
+                                service=service,
+                                instances=instances['instances'],
+                                git_sha=git_sha)
+                            if cluster_map[cluster]['deployed'] == len(cluster_map[cluster]['instances']):
+                                instance_csv = ", ".join(cluster_map[cluster]['instances'])
+                                print("Deploy to %s complete! (instances: %s)" % (cluster, instance_csv))
+                        bar.update(sum([v["deployed"] for v in cluster_map.values()]))
+                    if all([cluster['deployed'] == len(cluster["instances"]) for cluster in cluster_map.values()]):
+                        break
+                    else:
+                        time.sleep(10)
+    except TimeoutError:
+        human_status = ["{0}: {1}".format(cluster, data['deployed'])
+                        for cluster, data in cluster_map.items()]
+        _log(service=service, component='deploy',
+             line=("\nCurrent deployment status of {0} per cluster:\n"
+                   .format(deploy_group) + "\n".join(human_status)),
+             level='event')
+
+        line = ("\n\nTimed out after {0} seconds, waiting for {1} in {2} to be "
+                "deployed by PaaSTA. \n\n"
+                "This probably means the deploy hasn't suceeded. The new "
+                "service might not be healthy or one or more clusters could be "
+                "having issues.\n\n"
+                "To debug: try running 'paasta status -s {2} -vv' or 'paasta "
+                "logs -s {2}' to determine the cause.\n\n"
+                "{3} is still *marked* for deployment. To rollback, you can "
```

Comment:
What do you think about getting latest deployment, comparing it with the provided git_sha, and exiting when they don't match?
This would help to avoid waiting for deployment that was not marked, and these messages would be valid without any tweak, I guess.